File: Python/0006_zigzag.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if len(s) <= 1 or numRows <= 1:
            return s
        rt = str()
        for i in range(numRows):
            if i < len(s):
                rt += s[i]
            index = i
            while index < len(s):
                index0 = index + 2 * numRows - 2 - 2 * i
                index1 = index0 + 2 * i
                if index != index0 and index0 < len(s):
                    rt += s[index0]
                if index0 != index1 and index1 < len(s):
                    rt += s[index1]
                index = index1
        return rt

File: C++/0006_zigzag.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(s.length() <= 1 || numRows <= 1){ return s; }
        string rt = "";
        for(int i = 0; i < numRows; ++i){
            int index = i;
            if(index < s.length()){ rt += s[index]; }
            while(index < s.length()){
                int nextIndex0 = index + 2 * numRows - 2 - 2 * i;
                int nextIndex1 = nextIndex0 + 2 * i;
                if(index != nextIndex0 && nextIndex0 < s.length()){
                    rt += s[nextIndex0];
                }
                if(nextIndex0 != nextIndex1 && nextIndex1 < s.length()){
                    rt += s[nextIndex1];
                }
                index = nextIndex1;
            }
        }
        return rt;
    }
};
